audio/easytag/Portfile

```diff
@@ -62,10 +63,29 @@ variant nautilus description {Build nautilus context actions} {
     configure.args-delete   --disable-nautilus-actions
 }
 
+subport easytag-devel {
+    PortGroup       gitlab 1.0
+
+    conflicts       $name
+
+    version         20210912
+    gitlab.instance https://gitlab.gnome.org
+    gitlab.setup    GNOME easytag ed742959eb312355042ef0b5928b2bd5724298db
+    revision        0
+
+    fetch.type      git
```

Comment:
is it really necessary to do this? For example we do similar things often with the "github" PG and when getting it from a commit hash  does not require this (and one can use checksums).

Reply:
@reneeotten yes, it is. Without this hack I can't build this port:
```
--->  Computing dependencies for easytag-devel
--->  Fetching distfiles for easytag-devel
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://fra.de.distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://nue.de.distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://ftp-stud.hs-esslingen.de/pub/Mirrors/ftp.gnome.org/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://ftp.fau.de/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://ftp2.nluug.nl/windowing/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://ema.uk.distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://mse.uk.distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://mirrors.dotsrc.org/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://ftp1.nluug.nl/windowing/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://cph.dk.distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://fco.it.distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://ftp.acc.umu.se/pub/GNOME/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://www.mirrorservice.org/sites/ftp.gnome.org/pub/GNOME/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://mirror.umd.edu/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://mirror.cc.columbia.edu/pub/software/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://muug.ca/mirror/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://ftp.cse.buffalo.edu/pub/Gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://ywg.ca.distfiles.macports.org/mirror/macports/distfiles/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://ykf.ca.distfiles.macports.org/MacPorts/mpdistfiles/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://ftp.gnome.org/pub/GNOME/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://jog.id.distfiles.macports.org/macports/distfiles/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://jnb.za.distfiles.macports.org/distfiles/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://ftp.is.co.za/mirror/ftp.gnome.org/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://kmq.jp.distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://ftp.nara.wide.ad.jp/pub/X11/GNOME/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://cjj.kr.distfiles.macports.org/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://pek.cn.distfiles.macports.org/macports/distfiles/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://mirrors.ustc.edu.cn/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://aarnet.au.distfiles.macports.org/pub/macports/distfiles/easytag
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://mirror.internode.on.net/pub/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from https://fr2.rpmfind.net/linux/gnome.org/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://ftp.rpmfind.net/linux/gnome.org/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://www.gtlib.gatech.edu/pub/gnome/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from ftp://ftp.kddlabs.co.jp/pub/GNOME/sources/easytag/2.4
--->  Attempting to fetch easytag-ed742959eb312355042ef0b5928b2bd5724298db.tar.bz2 from http://atl.us.distfiles.macports.org/easytag
Error: Failed to fetch easytag-devel: The requested URL returned error: 404 Not Found
Error: See /opt/local/var/macports/logs/_Users_catap_src_macports-ports_audio_easytag/easytag-devel/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there is a bug.
Error: Processing of port easytag-devel failed

```

Reply:
the fact that this initial attempt doesn’t  work of course doesn’t mean that ‘git.fetch’ is the right thing to do. I haven’t investigated but if the PG works similarly to the “github” PG it should work. Or otherwise this PG should/could be probably improved…

Reply:
@reneeotten the PG `github` works on the same way. The root cause of issue is defined `master_sites` by this port.

Here I made different approach to use gitlab's `master_sites` for sub-port.

Reply:
> @reneeotten the PG `github` works on the same way. The root cause of issue is defined `master_sites` by this port.
>
> Here I made different approach to use gitlab's master_sites for sub-port.

I am fine with you using the gitlab PG for the subport, that's not the point I am trying to make here.

The "issue" is that when using the `github` PG you can specify a commit and then it will -just as it attempts here- download that as a `.tar.gz` for which you can add checksums and such and you don't need to do git checkout. That would be preferable here as well, but I haven't investigated if that's possible (i.e., if the `gitlab` PG and/or gitlab itself supports that).

Reply:
@reneeotten I've fixed an issue with no checksum.

The root cause of issue was defined `master_sites` at the port. When it is true, bot `github` and `gitlab` port group aren't worked. To fix them it is required to add `master_sites  ${gitlab.master_sites}` as I did.

After that it downloads `.tar.gz` from github and checks checksums.

Reply:
oh, sorry... no I see what you mean ;) It wasn't obvious to me from your earlier response and I didn't see the change you made in the Portfile. Thanks and yes, this looks good!